diff --git a/xbmc/utils/SystemInfo.cpp b/xbmc/utils/SystemInfo.cpp
index ee6311b..cf7fd8d 100644
--- a/xbmc/utils/SystemInfo.cpp
+++ b/xbmc/utils/SystemInfo.cpp
@@ -377,7 +377,21 @@ std::string CSysInfoJob::GetVideoEncoder()
 
 std::string CSysInfoJob::GetBatteryLevel()
 {
-  return StringUtils::Format("{}%", CServiceBroker::GetPowerManager().BatteryLevel());
+  std::ifstream myfile;
+  myfile.open("/tmp/battery.percent");
+  if ( !myfile.is_open() ) {
+    myfile.open("/sys/class/power_supply/battery/capacity");
+  }
+
+  std::string myline;
+  if ( myfile.is_open() ) {
+    while ( myfile ) {
+      std::getline (myfile, myline);
+      return myline + "%";
+    }
+  }
+
+  return "0%";
 }
 
 bool CSysInfoJob::SystemUpTime(int iInputMinutes, int &iMinutes, int &iHours, int &iDays)
